src/TestEnvironmenbt.Docker.Containers.Kafka/KafkaContainerWaiter.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using Confluent.Kafka.Admin;
+using Microsoft.Extensions.Logging;
+
+namespace TestEnvironment.Docker.Containers.Kafka
+{
+    public class KafkaContainerWaiter : BaseContainerWaiter<KafkaContainer>
+    {
+        private static readonly TimeSpan RetryDelay = TimeSpan.FromSeconds(5);
+
+        public KafkaContainerWaiter(ILogger logger = null)
+            : base(logger)
+        {
+        }
+
+        protected override async Task<bool> PerformCheck(KafkaContainer container, CancellationToken cancellationToken)
+        {
+            using var adminClient = new AdminClientBuilder(new AdminClientConfig
+                {
+                    BootstrapServers = container.GetUrl(),
+                })
+                .Build();
+            var topicName = $"{container.Name}_topic_health";
+            try
+            {
+                while (true)
```

Comment:
Please check base class here https://github.com/Deffiss/testenvironment-docker/blob/master/src/TestEnvironment.Docker/BaseContainerWaiter.cs

I'm not sure that you need to perform all these checks in loop by yourself on your level of 'waiting'. Try to simplify and make PerformCheck method as lightweight and fast as possible, like here - https://github.com/Deffiss/testenvironment-docker/blob/master/src/TestEnvironment.Docker.Containers.Postgres/PostgresContainerWaiter.cs

Reply:
I i simplified waiter